test/registry/FHIRRegistryClient.test.ts

```diff
@@ -0,0 +1,377 @@
+      fhirVersion: 'R4',
+      url: 'https://packages.simplifier.net/hl7.terminology.r4/1.1.2'
+    },
+    '1.1.1': {
+      name: 'hl7.terminology.r4',
+      version: '1.1.1',
+      description: 'None.',
+      dist: {
+        shasum: '1a1467bce19aace45771e0a51ef2ad9c3fe749821',
+        tarball: 'https://packages.simplifier.net/hl7.terminology.r4/1.1.1'
+      },
+      fhirVersion: 'R4',
+      url: 'https://packages.simplifier.net/hl7.terminology.r4/1.1.1'
+    }
+  }
+};
+
+describe('FHIRRegistryClient', () => {
+  const client = new FHIRRegistryClient('https://packages.fhir.org', { log: loggerSpy.log });
+  let axiosSpy: jest.SpyInstance;
+
+  describe('#constructor', () => {
+    it('should remove trailing slash from endpoint', async () => {
+      const clientWithSlash = new FHIRRegistryClient('https://packages.fhir.org/', {
+        log: loggerSpy.log
+      });
+      expect(clientWithSlash.endpoint).toBe('https://packages.fhir.org');
+    });
+  });
+
+  describe('#download', () => {
+    describe('#downloadSpecificVersion', () => {
+      beforeEach(() => {
+        loggerSpy.reset();
+      });
+      beforeAll(() => {
+        axiosSpy = jest.spyOn(axios, 'get').mockImplementation((uri: string): any => {
+          if (uri === 'https://packages.fhir.org/hl7.terminology.r4') {
+            return { data: TERM_PKG_RESPONSE };
+          } else if (uri === 'https://packages.fhir.org/hl7.terminology.r4/1.2.3-test') {
+            return {
+              status: 200,
+              data: Readable.from(['1.2.3-test-data'])
+            };
+          } else if (uri === 'https://packages.fhir.org/hl7.terminology.r4/2.2.2') {
+            return {
+              status: 200,
+              data: ''
+            };
+          } else if (uri === 'https://packages.fhir.org/hl7.terminology.r4/3.3.3') {
+            return {
+              status: 200
+            };
+          } else if (uri === 'https://packages.fhir.org/hl7.terminology.r4/5.5.5') {
+            return {
+              status: 'wrong-type',
+              data: Readable.from(['1.2.4-no-manifest-test-data'])
+            };
+          } else if (uri === 'https://packages.fhir.org/hl7.terminology.r4/5.5.6-test') {
+            return {
+              data: Readable.from(['5.5.6-test-data'])
+            };
+          } else if (uri === 'https://packages.fhir.org/hl7.terminology.r4/1.1.2') {
+            return {
+              status: 404,
+              data: Readable.from(['1.1.2-test-data'])
+            };
+          } else {
+            throw new Error('Not found');
+          }
+        });
+      });
+
+      afterAll(() => {
+        axiosSpy.mockRestore();
+      });
+
+      it('should throw error if no name given for download method', async () => {
+        const latest = client.download('', '5.5.5');
+        await expect(loggerSpy.getLastMessage('info')).toBe(
+          'Attempting to download #5.5.5 from https://packages.fhir.org//5.5.5'
+        );
```

Comment:
You don't need to await `loggerSpy.getLastMessage` because `loggerSpy.getLastMessage` is not an `async` function (and does not return a `Promise`). You can safely search for all cases of `await expect(loggerSpy` and remove the `await`. I think there are seven cases of this between this file and `NPMRegistryClient.test.ts`.